r_markdown/emissions.Rmd

```diff
@@ -457,49 +556,33 @@ ___
   head(ions, 2)
 ```
 
-## carbonyls
-
-* load data
-
-```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
-
-```
+# carbonyls
 
 * select data
 
 ```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
   carbonyls <- dplyr::select(carbonyls_merged,
                              id, qc,
-                             pre_flow, post_flow,
-                             time_start, time_end,
+                             flow,
+                             start, end,
                              pol, mass_ug,
                              conc_bg) %>%
-               dplyr::filter(grepl("^[A-Z]", id) == FALSE) %>%
-               dplyr::rename(start = time_start,
-                             end = time_end)
+               dplyr::filter(grepl("^[A-Z]", id) == FALSE)
 ```
 
 * calculate concentrations
 
 ```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
   carbonyls <- dplyr::mutate(carbonyls, dur = (end - start) / 60,
-                             flow = (pre_flow + post_flow) / 2,
-                             conc = (mass_ug * 1000 / (flow * dur)) - conc_bg)
+                             conc = (mass_ug * 1000 / (flow * dur)))
 ```
 
 * calculate mass emitted
 
 ```{r, echo=TRUE, warning=FALSE, message=FALSE, results='markup', cache=FALSE}
   carbonyls <- dplyr::mutate(carbonyls, 
                              mass_emitted = (conc - conc_bg) * flow_hood * dur,
-                             mass_carbon = mass_emitted * 0)  # carbonyls are gases
-
-  carbonyls_g <- dplyr::group_by(carbonyls, id) %>%
-                 dplyr::filter(!is.na(mass_emitted), !is.na(mass_carbon)) %>%
-                 dplyr::summarise(mass_emitted = sum(mass_emitted),
-                                  mass_carbon = sum(mass_carbon),
-                                  qc = first(qc)) %>%
-                 dplyr::mutate(pol = "total_carbonyls")
+                             mass_carbon = mass_emitted * 0.95) # assume 95% carbon
```

Comment:
Can you just add the molecular weights and number of carbon atoms for the carbonyls?